File: workbooks/js-to-postgresql/scripts/seed.js
import { createPool } from "./db.js";
import {
  setupUsersTable,
  insertUsers,
  countUsers,
  usersTableSchema,
  users,
} from "./seeds/users.js";
import {
  setupCustomersTable,
  insertCustomers,
  countCustomers,
  customersTableSchema,
  customers,
  setupOrdersTable,
  insertOrders,
  countOrders,
  ordersTableSchema,
  orders,
  setupOrderItemsTable,
  insertOrderItems,
  countOrderItems,
  orderItemsTableSchema,
  orderItems,
  createIndexes,
  joinAcrossAllTables,
} from "./seeds/customerOrders.js";

async function seedUsers() {
  const pool = createPool();

  await setupUsersTable(pool);
  await insertUsers(pool, users);
  const count = await countUsers(pool);
  console.log(`Seeded ${count} users`);
}

async function seedCustomers() {
  const pool = createPool();
  await setupCustomersTable(pool);
  await insertCustomers(pool, customers);
  const count = await countCustomers(pool);
  console.log(`Seeded ${count} customers`);
}

async function seedOrders() {
  const pool = createPool();
  await setupOrdersTable(pool);
  await insertOrders(pool, orders);
  const count = await countOrders(pool);
  console.log(`Seeded ${count} orders`);
}

async function seedOrderItems() {
  const pool = createPool();
  await setupOrderItemsTable(pool);
  await insertOrderItems(pool, orderItems);
  const count = await countOrderItems(pool);
  console.log(`Seeded ${count} order items`);
  await createIndexes(pool);
  const rows = await joinAcrossAllTables(pool);
  console.log(rows);
}

async function seed() {
  await seedUsers();
  await seedCustomers();
  await seedOrders();
  await seedOrderItems();

}

seed().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: workbooks/python-to-postgresql/scripts/seed.py
import sys

from scripts.db import get_connection
from scripts.seeds.users import (
    USERS,
    count_users,
    insert_users,
    setup_users_table,
)
from scripts.seeds.customer_orders import (
    CUSTOMERS,
    ORDER_ITEMS,
    ORDERS,
    count_customers,
    count_order_items,
    count_orders,
    create_indexes,
    insert_customers,
    insert_order_items,
    insert_orders,
    join_across_all_tables,
    setup_customers_table,
    setup_order_items_table,
    setup_orders_table,
)


def seed_users():
    conn = get_connection()
    try:
        setup_users_table(conn)
        insert_users(conn, USERS)
        count = count_users(conn)
        print(f"Seeded {count} users")
    finally:
        conn.close()


def seed_customers():
    conn = get_connection()
    try:
        setup_customers_table(conn)
        insert_customers(conn, CUSTOMERS)
        count = count_customers(conn)
        print(f"Seeded {count} customers")
    finally:
        conn.close()


def seed_orders():
    conn = get_connection()
    try:
        setup_orders_table(conn)
        insert_orders(conn, ORDERS)
        count = count_orders(conn)
        print(f"Seeded {count} orders")
    finally:
        conn.close()


def seed_order_items():
    conn = get_connection()
    try:
        setup_order_items_table(conn)
        insert_order_items(conn, ORDER_ITEMS)
        count = count_order_items(conn)
        print(f"Seeded {count} order items")
        create_indexes(conn)
        rows = join_across_all_tables(conn)
        print(rows)
    finally:
        conn.close()


def seed():
    seed_users()
    seed_customers()
    seed_orders()
    seed_order_items()


if __name__ == "__main__":
    try:
        seed()
    except Exception as err:
        print(err, file=sys.stderr)
        sys.exit(1)
